### PR TITLE
PRL-2444:created new pr with refactored changes

### PR DESCRIPTION
**Pull request by khurshedk**
### JIRA link (if applicable) ###
PRL-2444

### Change description ###

created new pr with refactored changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java [hunk 1]
```diff
@@ -106,6 +111,42 @@ public void sendEmailToApplicantAndRespondent(CaseDetails caseDetails) {
...
+        if (caseData.getState() == State.ALL_FINAL_ORDERS_ISSUED) {
+            sendNotificationToRespondentSolicitor(caseDetails);
+            sendNotificationToRespondent(caseDetails);
+        }
+    }
+
+    private void sendNotificationToRespondent(CaseDetails caseDetails) {
+        CaseData caseData = emailService.getCaseData(caseDetails);
+
+        for (Element<PartyDetails> respondent : caseData.getRespondents()) {
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:125]: what about DA cases?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java [hunk 2]
```diff
@@ -106,6 +111,42 @@ public void sendEmailToApplicantAndRespondent(CaseDetails caseDetails) {
...
+                EmailTemplateNames.CA_CITIZEN_RES_NOTIFICATION,
+                buildRespondentEmail(caseDetails, respondent.getValue()),
+                LanguagePreference.english
+            );
+        }
+    }
+
+    private void sendNotificationToRespondentSolicitor(CaseDetails caseDetails) {
+
+        for (Map<String, List<String>> resSols : getRespondentSolicitor(caseDetails)) {
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:137]: If there are multiple respondent solicitors?

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java [hunk 3]
```diff
@@ -174,7 +238,21 @@ private List<PartyDetails> getRespondents(CaseData caseData) {
             .collect(Collectors.toList());
     }
 
-
+    private List<Map<String, List<String>>> getRespondentSolicitor(CaseDetails caseDetails) {
+        CaseData caseData = emailService.getCaseData(caseDetails);
+        return caseData
+            .getRespondents()
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:244]: DA case check here?

**Review by shashisk97 — COMMENTED** [review]
Please test the flow for both CA and DA cases

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java (hunk 3 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:254]: The requirement to send emails to the Respondent solicitor was only for CA applications, so DA applications are not catered to here. please refer PRL-1974

File: src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java (hunk 2 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderEmailService.java:147]: we are getting list of solicitor inside map,so I believe we will get all respondent solictors information

**Review by AlokDatta — APPROVED** [review]

**Pull request merged**